src/main/java/org/commcare/formplayer/application/FormSubmissionController.java

```diff
@@ -252,38 +275,100 @@ private SubmitResponseBean processFormXml(FormSubmissionContext context) throws
         return context.success();
     }
 
+    private MultiValueMap<String, HttpEntity<Object>> getMultiPartFormBody(FormSession formSession)
+            throws IOException {
+        MultiValueMap<String, HttpEntity<Object>> body = new LinkedMultiValueMap<>();
+
+        // Add any media files associated with the form
+        Path mediaDirPath = formSession.getMediaDirectoryPath(restoreFactory.getDomain(),
+                restoreFactory.getUsername(), restoreFactory.getAsUsername(), storageFactory.getAppId());
+        File mediaFile = mediaDirPath.toFile();
+        if (mediaFile.exists()) {
+            File[] files = Objects.requireNonNull(mediaDirPath.toFile().listFiles());
+            if (files.length > maxAttachmentsPerForm) {
+                failWithError("form.upload.attachments.threshold.error.message", maxAttachmentsPerForm.toString());
```

Comment:
nit: update locale message to be more concise and consistent with the other error messages:
```suggestion
                failWithError("form.upload.attachments.limit_exceeded", maxAttachmentsPerForm.toString());
```